packages/go_router/test/on_exit_test.dart

```diff
@@ -25,7 +25,7 @@ void main() {
             path: '1',
             builder: (BuildContext context, GoRouterState state) =>
                 DummyScreen(key: page1),
-            onExit: (BuildContext context) {
+            onExit: (BuildContext context, GoRouterState state) {
```

Comment:
Can you add a new test that assert the properties in state, to make sure the state is accurate with the route?